src/rootcheck/check_rc_if.c

```diff
@@ -57,10 +57,9 @@ void check_rc_if()
 {
     int _fd, _errors = 0, _total = 0;
     struct ifreq tmp_str[16];
-
+    const int max_if = sizeof(tmp_str)/sizeof(struct ifreq);
```

Comment:
Hi @RamiroRD 

I think the behavior is the same here because the `_if.ifc_len` variable was initialized with `sizeof(tmp_str). This `sizeof` contains the size of 16 elements of the array in bytes. Then it uses the _ifend pointer to get the last address of the array. I found the way to read the vector very complicated and I made it simpler.

Best regards
Hanes